apps/frontend/src/components/TimeCardPage/SubmitCard.tsx

```diff
@@ -1,61 +1,214 @@
-import React, { useState, useEffect } from 'react'
-import { WeeklyCommentModal } from './CommentModal';
-import { Box, Card, CardHeader, CardBody, CardFooter, Button } from '@chakra-ui/react';
-import { CardState } from './types'
-import { CommentSchema } from 'src/schemas/RowSchema';
-
-interface SubmitCardProps {
-    setWeeklyComments: Function;
-    setWeeklyReports: Function;
-    weeklyComments: CommentSchema[];
-    weeklyReports: CommentSchema[];
+import { CardState } from "./types";
+
+import React, { useState, useEffect, useContext } from "react";
+import { Box, Card, CardBody, CardFooter, Button } from "@chakra-ui/react";
+import { DEFAULT_COLORS } from "src/constants";
+import ApiClient from "src/components/Auth/apiClient";
+import * as updateSchemas from "src/schemas/backend/UpdateTimesheet";
+import { StatusType, StatusEntryType } from "src/schemas/StatusSchema";
+import { UserTypes } from "./types";
+import moment from "moment";
+import { useToast } from "@chakra-ui/react";
+import { UserContext } from "./UserContext";
+import { TimesheetStatus } from "src/schemas/TimesheetSchema";
+
+interface submitCardProps {
+  timesheetId: number;
+  associateId: string;
+  timesheetStatus: StatusType;
+  refreshTimesheetCallback: Function;
 }
 
-export default function SubmitCard({
-    setWeeklyComments,
-    setWeeklyReports,
-    weeklyComments,
-    weeklyReports
-}: SubmitCardProps) {
+export default function SubmitCard(props: submitCardProps) {
+  const toast = useToast();
+  const currUser = useContext(UserContext);
 
+  /** Whether or not the logged-in user has submitted this timesheet yet.*/
   const [submitted, setSubmitted] = useState(false);
+
+  /** The date and time (as a moment) that the logged-in user submitted/reviewed/finalized this timesheet.*/
   const [submitDate, setSubmitDate] = useState(null);
```

Comment:
nit: submitMoment, setSubmitMoment

Reply:
actually I think this isn't even getting used, going to remove it